build/v1/types.go

```diff
@@ -1008,7 +1008,32 @@ type ImageChangeTrigger struct {
 
 // BuildTriggerPolicy describes a policy for a single trigger that results in a new Build.
 type BuildTriggerPolicy struct {
-	// type is the type of build trigger
+	// type is the type of build trigger. Valid values:
+	//
+	// - GitHub
+	// GitHubWebHookBuildTriggerType represents a trigger that launches builds on
+	// GitHub webhook invocations
+	//
+	// - Generic
+	// GenericWebHookBuildTriggerType represents a trigger that launches builds on
+	// generic webhook invocations
+	//
+	// - GitLab
+	// GitLabWebHookBuildTriggerType represents a trigger that launches builds on
+	// GitLab webhook invocations
+	//
+	// - Bitbucket
+	// BitbucketWebHookBuildTriggerType represents a trigger that launches builds on
+	// Bitbucket webhook invocations
+	//
+	// - ImageChange
+	// ImageChangeBuildTriggerType represents a trigger that launches builds on
+	// availability of a new version of an image
+	//
+	// - ConfigChange
+	// ConfigChangeBuildTriggerType will trigger a build on an initial build config creation
+	// WARNING: In the future the behavior will change to trigger a build on any config change
+	//
```

Comment:
Nit - remove the trailing comment line.

I realize the warning is in the type's godoc, but we haven't changed the behavior of the ConfigChange trigger since 3.11, and I think there are valid reasons why we may never get this trigger past "start a build after creating this BuildConfig".

cc @bparees - you may recall exactly why we don't trigger on any config change.

Reply:
we just never implemented it, basically.  Implementing it requires that we find a way to be level triggered instead of edge triggered.  One challenge is that we have some status information in our spec fields, so ensuring we only trigger builds on legitimate spec changes would require some extra thought.  But fundamentally it just never became an important feature need.  No other reason I can recall.

Reply:
@adambkaplan updated

Reply:
@alvaroaleman given Ben's comment, please restore the warning. That last empty line in the godoc comment can be removed.

Reply:
@adambkaplan updated